src/calchartdoc.cpp

```diff
@@ -279,6 +281,25 @@ wxInputStream& CalChartDoc::LoadObject(wxInputStream& stream)
 }
 #endif
 
+bool CalChartDoc::exportViewerFile(const wxString& filepath) {    
+    JSONElement mainObjectElement = JSONElement::makeObject();
+    
+    JSONDataObjectAccessor mainObject = mainObjectElement;
```

Comment:
A `JSONDataObjectAccessor` carries only a reference to a `JSONElement`, so the `JSONElement` needs to exist outside of the accessor.